tests/test_10_base.py

```diff
@@ -15,66 +11,46 @@ class BaseNameTest(Archive):
     _mimes = ["mime1", "mime2"]
 
 
-class Archive(unittest.TestCase):
-    def test_10_guess_basename(self):
-        fileobj = BytesIO(b"")
-        try:
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime2", "xxx", fileobj)
-            )
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime1", "xxx.ext2", fileobj)
-            )
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("mime2", "xxx.ext1", fileobj)
-            )
-        except ValueError as e:
-            self.fail(repr(e))
-        try:
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("xxx", "xxx.ext1", fileobj)
-            )
-        except ValueError:
-            pass
-        else:
-            self.fail("guessing should has failed")
-
-
-class ArchiveFileTest(unittest.TestCase):
-    def _regular_tests(self, archive, fileobj, filename, text):
-        self.assertEqual(
-            archive.fileno(), fileobj.fileno(), "file no does not match!"
-        )
-        self.assertEqual(
-            archive.name, filename, "name attribute does not match!"
-        )
+def test_10_guess_basename():
```

Comment:
This test before belonged to Archive so I guess it's testing something related to the Archive type.

Now, it's a standalone test case so I don't know which type it's supposed to be testing.

This makes test incredibly harder to understand, to me. Maybe we should use a different naming convention for pytest tests? What is the best practice to organize tests when using this library?

Reply:
That's an important point. `pytest` supports classes but can work without them as well.

There are three levels of tests: modules, classes, and function. I deliberately removed the level of classes, but now we have two possibilities: 1. return them back and nothing has changed, or 2. rename the methods to include the level into them. Which one do you prefer? Would you like to prepare a system?

Reply:
I think you should just rename it to something that makes sense and remote the `test_xx_` prefix (it's useless).

For example, this test makes sure that the `_guess` method can strip the extension from the filename. Maybe it could be named `test_extension_is_stripped_from_file_name`

Reply:
I usually keep the structure of tests very close to that of the code so, if you know about this, it's very easy to understand where are the tests of what.

I must be unusually meticulous about this and it may not have enough value for smaller modules. :)